pyiron_base/project/maintenance.py

```diff
@@ -69,6 +72,37 @@ def get_repository_status():
         return report
 
 
+class LocalMaintenance:
+    def __init__(self, project):
+        self._project = project
+
+        from pyiron_base import Project
+
+        self._project = Project()
+
+    def defragment_storage(
+        self,
+        recursive: bool = True,
+        progress: bool = True,
+        **kwargs: dict,
+    ):
+        """
+        Iterate over the jobs within the current project and it is sub projects and rewrite the hdf file
+
+        Args:
+            recursive (bool): search subprojects [True/False] - True by default
+            progress (bool): if True (default), add an interactive progress bar to the iteration
+            **kwargs (dict): Optional arguments for filtering with keys matching the project database column name
+                            (eg. status="finished"). Asterisk can be used to denote a wildcard, for zero or more
+                            instances of any character
+        """
+        for job in self._project.iter_jobs(
+            recursive=recursive, progress=progress, convert_to_object=False, **kwargs
+        ):
+            hdf = job.project_hdf5
+            hdf.rewrite_hdf(job.name)
```

Comment:
Would it be a good idea to remove the `job_name` argument for `rewrite_hdf` in this PR itself?

Reply:
I am not really sure. Looking at `rewrite_hdf` I sort of had the feeling that this _only_ rewrites the job itself and not (necessarily) the overall file. I wanted to do (and probably include some tests into base) to see if it actually correctly rewrites the whole hdf file if multiple jobs are in there or has a latent bug for such cases.